lms/views/dashboard.py

```diff
@@ -17,10 +18,29 @@ def __init__(self, request) -> None:
         request_method="POST",
     )
     def assignment_redirect_from_launch(self):
+        """
+        Entry point to the dashboards from an LTI launch.
+
+        Here we "promote" the LTILaunch token present as a form parameter to a cookie.
+        """
         assignment_id = self.request.matchdict["id_"]
-        return HTTPFound(
+        response = HTTPFound(
             location=self.request.route_url("dashboard.assignment", id_=assignment_id),
         )
+        # Encode the current LTIUser as a cookie
+        auth_token = (
+            BearerTokenSchema(self.request).authorization_param(self.request.lti_user)
+            # White space is not allowed as a cookie character, remove the leading part
+            .replace("Bearer ", "")
+        )
+        response.set_cookie(
+            "authorization",
+            value=auth_token,
+            secure=not self.request.registry.settings["dev"],
+            httponly=True,
+            max_age=60 * 60 * 24 * 30,  # 30 Days
```

Comment:
We'll have to revisit this. Added a note on the project.

Reply:
Since we have discussed the token expires in one day, should we set the same expiration to the cookie?

Reply:
Done :+1: